dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -226,27 +189,25 @@ def _get_regressor_object(self, action, func_args):
         else:
             raise ValueError("The function: {} is not supported by dowhy at the moment".format(action))
 
-    def _permute(self, new_data, permute_fraction):
-        new_data = pd.DataFrame(new_data)
-        new_data.columns = ['y']
-
-        if permute_fraction == 1:
-            return new_data['y'].sample(frac=1).values
-        else: 
-            return new_data.apply( lambda row: self.replace(row, new_data, permute_fraction), axis=1).values
-        
-
-    def replace(self, row, new_data, fraction):
+    def _permute(self, new_outcome, permute_fraction):
         '''
         In this function, we make use of the Fisher Yates shuffle:
         https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
-        '''  
-        if np.random.uniform(0,1) <= fraction:
-            index = np.random.randint(row.name + 1, len(new_data))
-        else:
-            index = row.name
-        
-        return new_data['y'][index]
-    
-    def _noise(self, new_data, std_dev):
-        return new_data + np.random.randn(new_data.shape[0]) * std_dev
\ No newline at end of file
+        '''
+        if permute_fraction == 1:
+            new_outcome = pd.DataFrame(new_outcome)
+            new_outcome.columns = ['y']
+            return new_outcome['y'].sample(frac=1).values
+        else: 
+            changes = np.where( np.random.uniform(0,1,new_outcome.shape[0]) <= permute_fraction )[0] # As this is tuple containing a single element (array[...])

```

Comment:
funny thing: let's say permute fraction is 0.5. then because of swapping which changes two items at a time, won't we have changed the index of every element?

Reply:
That's true. In that case, we should we halve the threshold?

Reply:
yes I think so. you can halve it in the "if" statement.